crates/analyzer/src/db/queries/types.rs

```diff
@@ -69,3 +69,45 @@ pub fn type_alias_type_cycle(
 
     Analysis::new(err, context.diagnostics.take().into())
 }
+
+pub fn is_zero_sized(db: &dyn AnalyzerDb, ty: types::TypeId) -> bool {
+    let typ = db.lookup_intern_type(ty);
+    match typ {
+        Type::Base(base) => matches!(base, Base::Unit),
+        Type::Array(Array { size, inner }) => size == 0 || inner.is_zero_sized(db),
+        Type::Tuple(Tuple { items }) => {
+            for item in items.iter() {
+                if !db.is_zero_sized(*item) {
+                    return false;
+                }
+            }
+            true
+        }
+        Type::String(FeString { max_size }) => max_size == 0,
+        Type::Struct(struct_id) => {
+            for field_type_id in db.struct_all_fields(struct_id).iter() {
+                // assume all type is correct.
+                let field_type = db.struct_field_type(*field_type_id).value.unwrap();
+                if !(field_type).is_zero_sized(db) {
+                    return false;
+                }
+            }
+            true
+        }
+        Type::Contract(contract_id) => {
+            for field_type_id in db.contract_all_fields(contract_id).iter() {
```

Comment:
Logic to handle struct and contract very similar. Have any better way to refactor it....

Reply:
A contract type is really just an address (its size is that of the `address` type), so you don't need to look at the fields.

Reply:
> A contract type is really just an address (its size is that of the `address` type), so you don't need to look at the fields.

Thanks! I fixed it